paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/TableController.java

```diff
@@ -88,6 +112,276 @@ public R<Void> createTable(@RequestBody TableInfo tableInfo) {
+                                        DataTypeConvertUtils.convert(
+                                                new PaimonDataType(
+                                                        tableColumn.getDataType(),
+                                                        tableColumn.isNullable(),
+                                                        tableColumn.getLength0(),
+                                                        tableColumn.getLength1())))
+                                .comment(tableColumn.getComment())
+                                .kind(OperatorKind.ADD_COLUMN)
+                                .build();
+                entityList.add(alterTableEntity);
+            }
+            TableManager.alterTable(
+                    catalog, tableInfo.getDatabaseName(), tableInfo.getTableName(), entityList);
+            if (options.size() > 0) {
+                TableManager.setOptions(
+                        catalog, tableInfo.getDatabaseName(), tableInfo.getTableName(), options);
+            }
+            return R.succeed();
+        } catch (Exception e) {
+            e.printStackTrace();
+            return R.failed(Status.TABLE_ADD_COLUMN_ERROR);
+        }
+    }
+
+    /**
+     * Drops a column from a table.
+     *
+     * @param tableInfo The information of the table.
+     * @return The result of the operation.
+     */
+    @PostMapping("/dropColumn")
+    public R<Void> dropColumn(@RequestBody TableInfo tableInfo) {
+        try {
+            Catalog catalog = CatalogUtils.getCatalog(getCatalogInfo(tableInfo.getCatalogName()));
+            List<TableColumn> tableColumns = tableInfo.getTableColumns();
+            List<AlterTableEntity> entityList = new ArrayList<>();
+            for (TableColumn tableColumn : tableColumns) {
+                AlterTableEntity alterTableEntity =
+                        AlterTableEntity.builder()
+                                .columnName(tableColumn.getField())
+                                .kind(OperatorKind.DROP_COLUMN)
+                                .build();
+                entityList.add(alterTableEntity);
+            }
+            TableManager.alterTable(
+                    catalog, tableInfo.getDatabaseName(), tableInfo.getTableName(), entityList);
+            return R.succeed();
+        } catch (Exception e) {
+            e.printStackTrace();
+            return R.failed(Status.TABLE_DROP_COLUMN_ERROR);
+        }
+    }
+
+    /**
+     * Renames a column in a table.
+     *
+     * @param tableInfo The information of the table.
+     * @return The result of the operation.
+     */
+    @PostMapping("/renameColumn")
+    public R<Void> renameColumn(@RequestBody TableInfo tableInfo) {
+        try {
+            Catalog catalog = CatalogUtils.getCatalog(getCatalogInfo(tableInfo.getCatalogName()));
+            List<TableColumn> tableColumns = tableInfo.getTableColumns();
+
+            if (tableColumns.size() != 2) {
```

Comment:
Why renaming column require two table columns objects? How does the user solve this problem?